src/Layer.js

```diff
@@ -305,8 +350,49 @@ export default class Layer {
         return this._integrator;
     }
 
+    /**
+     *  Get the features in a position
+     * @param  {object} pos
+     * @param  {number} pos.x
+     * @param  {number} pos.y
+     * @memberof carto.Layer
+     * @instance
+     * @api
+     */
     getFeaturesAtPosition (pos) {
-        return this._renderLayer.getFeaturesAtPosition(pos).map(this._addLayerIdToFeature.bind(this));
+        return this.visible
+            ? this._renderLayer.getFeaturesAtPosition(pos).map(this._addLayerIdToFeature.bind(this))
+            : [];
+    }
+
+    /**
+     * Change layer visibility to visible
+     *
+     * @memberof carto.Layer
+     * @instance
+     * @api
+     *
+     * @fires updated
+     */
+    show () {
+        this.visible = true;
```

Comment:
If `visible` is part of the API, then it should have a setter, if not it should be `_visible`